sdk/monitor/monitor_client.go

```diff
@@ -43,22 +53,37 @@ func NewClient(cc *grpc.ClientConn, eventFactory EventFactory, streamConstructor
 	}
 
 	errorChannel := make(chan error, 1)
-	eventChannel := make(chan Event, 1)
+	eventChannel := make(chan Event, clientEventCapacity)
 	go func() {
 		defer close(eventChannel)
 		defer close(errorChannel)
 
-		for {
-			message, err := stream.Recv()
-			if err != nil {
-				errorChannel <- err
-				return
+		recvCh := make(chan recvResult)
+		go func() {
+			for {
+				message, err := stream.Recv()
+				recvCh <- recvResult{msg: message, err: err}
 			}
+		}()
 
-			if event, err := eventFactory.EventFromMessage(context.Background(), message); err != nil {
-				logrus.Errorf("An error during conversion event: %v", err)
-			} else {
-				eventChannel <- event
+		for {
+			select {
+			case r := <-recvCh:
+				if r.err != nil {
+					logrus.Infof("An error during recv: %v", r.err)
+					errorChannel <- r.err
+					return
+				}
+				if event, err := eventFactory.EventFromMessage(context.Background(), r.msg); err != nil {
```

Comment:
Why `eventFactory` needs `context` and is it ok to pass `context.Background` here?

Reply:
Just move this code, please take a look at current implementation